packages/flutter_tools/lib/src/android/android_workflow.dart

```diff
@@ -249,19 +250,14 @@ class AndroidLicenseValidator extends DoctorValidator {
     final Future<void> output = process.stdout
       .transform<String>(const Utf8Decoder(allowMalformed: true))
       .transform<String>(const LineSplitter())
-      .listen(_onLine)
+      .listen(_handleLine)
       .asFuture<void>(null);
     final Future<void> errors = process.stderr
       .transform<String>(const Utf8Decoder(allowMalformed: true))
       .transform<String>(const LineSplitter())
-      .listen(_onLine)
+      .listen(_handleLine)
       .asFuture<void>(null);
-    try {
-      await Future.wait<void>(<Future<void>>[output, errors]).timeout(const Duration(seconds: 30));
-    } catch (TimeoutException) {
-      printTrace('Intentionally killing ${androidSdk.sdkManagerPath}');
-      processManager.killPid(process.pid);
-    }
+    await Future.wait<void>(<Future<void>>[output, errors]);
```

Comment:
Here's a case where we won't notify the user in any meaningful way if the process hangs.  We should warn if it takes too long to complete, or show a progress indicator.

Reply:
in this case, different versions of the Android SDK could produce different output and expect different input, causing us to hang for all instances of Android SDK version X.

Reply:
I've made `flutter doctor` be explicit about what's taking a long time if it takes a long time.